### DISCUSSION TITLE
Caching and fallbackData are not working

### DISCUSSION THREAD

**Discussion by fxOne** [vercel/swr#2561] [Q&A]
# Bug report

## Description / Observed Behavior

The caching and fallback data in the [server-render example](https://github.com/vercel/swr/tree/main/examples/server-render) are not working as expected. I can see multiple requests to `pokeapi.co` which shouldn't be there.

![image](https://user-images.githubusercontent.com/5929640/231569729-7174243d-9508-4eaa-9a8a-1073f22f6a2c.png)

## Expected Behavior

I would expect to see no request to pokeapi.co on the initial request as it is rendered on server and no further request on switching between the pages as the cache hits.

## Repro Steps / Code Example

https://github.com/vercel/swr/tree/main/examples/server-render

## Additional Context

SWR version: latest

**Comment by koba04** [comment]
@fxOne 
That's an expected behavior, SWR revalidates even if `fallbackData` is provided.
You can disable the behavior by using `useSWRImmutable` instead of `useSWR`.

https://swr.vercel.app/docs/revalidation#disable-automatic-revalidations

**Reply by ADTC** [reply]
Hey it would be nice if the docs mention it: https://swr.vercel.app/docs/api#options

* `fallbackData`: initial data to be returned (Note: This is per-hook. To avoid revalidating, [mark it as immutable](https://swr.vercel.app/docs/revalidation#disable-automatic-revalidations).)
